### ISSUE TITLE
NullPointerException on RedisOperationsSessionRepository cleanupPrincipalIndex

### ISSUE THREAD

**Issue by madegroot** [spring-projects/spring-session#1063]
I'm encountering an NullPointerException in the RedisOperationsSessionRepository.
From reading the code, in this case a null value is expected, but a getter is invoked on the object a flew lines later.  

using Spring-Session: 1.3.2

```
java.lang.NullPointerException: null
	at org.springframework.session.data.redis.RedisOperationsSessionRepository.cleanupPrincipalIndex(RedisOperationsSessionRepository.java:544)
	at org.springframework.session.data.redis.RedisOperationsSessionRepository.onMessage(RedisOperationsSessionRepository.java:530)
	at org.springframework.data.redis.listener.RedisMessageListenerContainer.executeListener(RedisMessageListenerContainer.java:249)
	at org.springframework.data.redis.listener.RedisMessageListenerContainer.processMessage(RedisMessageListenerContainer.java:239)
	at org.springframework.data.redis.listener.RedisMessageListenerContainer$1.run(RedisMessageListenerContainer.java:967)
	at java.lang.Thread.run(Thread.java:745)
```

Class: org.springframework.session.data.redis.RedisOperationsSessionRepository
Method: onMessage

Line: 521 states `if (session == null)`
Line: 530 calls `cleanupPrincipalIndex(session)`
Line: 544 calls the getter on the null object: `session.getId();`

I'd expect that null pointers are be prevented.

**Comment by vpavic** [comment]
Thanks for the report @madegroot. This was fixed recently in #1049 and will be a part of `1.3.3.RELEASE`.

Duplicate of #1049

**Comment by ipofss** [comment]
Exception in thread "redisMessageListenerContainer-2" java.lang.NullPointerException
	at redis.clients.jedis.BinaryJedisPubSub.psubscribe(BinaryJedisPubSub.java:53)
	at org.springframework.data.redis.connection.jedis.JedisSubscription.doPsubscribe(JedisSubscription.java:49)
	at org.springframework.data.redis.connection.util.AbstractSubscription.pSubscribe(AbstractSubscription.java:139)
	at org.springframework.data.redis.listener.RedisMessageListenerContainer$SubscriptionTask$PatternSubscriptionTask.run(RedisMessageListenerContainer.java:707)
	at java.lang.Thread.run(Thread.java:745)

It happened to me, too, But just once. When I shutdown the project, and then start the project, I saw it.
I use Spring-Boot framework, and spring-session.
`<parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>2.1.2.RELEASE</version>
        <relativePath/> 
    </parent>`
`<dependency>
            <groupId>org.springframework.session</groupId>
            <artifactId>spring-session-data-redis</artifactId>
        </dependency>`
